rust/src/dcerpc/dcerpc.rs

```diff
@@ -892,43 +1057,51 @@ pub extern "C" fn rs_dcerpc_state_transaction_free(_state: *mut std::os::raw::c_
 pub extern "C" fn rs_dcerpc_get_tx_detect_state(
     vtx: *mut std::os::raw::c_void,
 ) -> *mut core::DetectEngineState {
-    let dce_state = cast_pointer!(vtx, DCERPCState);
-    match dce_state.de_state {
+    let dce_tx = cast_pointer!(vtx, DCERPCTransaction);
+    match dce_tx.de_state {
         Some(ds) => ds,
         None => std::ptr::null_mut(),
     }
 }
 
 #[no_mangle]
 pub extern "C" fn rs_dcerpc_set_tx_detect_state(
-    vtx: *mut std::os::raw::c_void,
-    de_state: *mut core::DetectEngineState,
+    vtx: *mut std::os::raw::c_void, de_state: *mut core::DetectEngineState,
 ) -> u8 {
-    let dce_state = cast_pointer!(vtx, DCERPCState);
-    dce_state.de_state = Some(de_state);
+    let dce_tx = cast_pointer!(vtx, DCERPCTransaction);
+    dce_tx.de_state = Some(de_state);
     0
 }
 
 #[no_mangle]
 pub extern "C" fn rs_dcerpc_get_tx(
-    state: *mut std::os::raw::c_void,
-    _tx_id: u64,
-) -> *mut DCERPCState {
-    let dce_state = cast_pointer!(state, DCERPCState);
-    dce_state
+    vtx: *mut std::os::raw::c_void, tx_id: u32,
+) -> *mut DCERPCTransaction {
+    let dce_state = cast_pointer!(vtx, DCERPCState);
+    match dce_state.get_tx(tx_id) {
+        Some(tx) => tx,
+        None => std::ptr::null_mut(),
+    }
 }
 
 #[no_mangle]
-pub extern "C" fn rs_dcerpc_get_tx_cnt(_state: *mut std::os::raw::c_void) -> u8 {
-    1
+pub extern "C" fn rs_dcerpc_get_tx_cnt(vtx: *mut std::os::raw::c_void) -> u8 {
+    let dce_state = cast_pointer!(vtx, DCERPCState);
+    dce_state.transactions.len() as u8
```

Comment:
this should actually be returning a `max_id` instead of a current count of the txs (bad api name)